silq/analysis/analysis.py

```diff
@@ -715,8 +717,44 @@ def analyse_EPR(empty_traces: np.ndarray,
             'mean_high_blip_duration': results_read['mean_high_blip_duration']}
 
 
-def analyse_flips(up_proportions_arrs: List[np.ndarray],
-                  threshold_up_proportion: Union[Sequence, float]):
+def analyse_threshold_up_proportion(up_proportions_arrs: np.ndarray,
```

Comment:
Am I correct in thinking that this adaptive threshold up proportion is only for the case where there is a single up_proportion array? If so, this should be written in the function docstring, and maybe the function name should also reflect this.  `analyse_threshold_up_proportion` should be replaced with `determine_threshold_up_proportion`, since you're analysing traces to determine the threshold.